server/app/services/PaginationSpec.java

```diff
@@ -1,23 +1,30 @@
 package services;
 
+import java.util.Optional;
+
 /** Specifies pagination behavior for a paginated query. */
 public class PaginationSpec {
 
   public static PaginationSpec MAX_PAGE_SIZE_SPEC = new PaginationSpec(Integer.MAX_VALUE, 1);
 
-  private final int pageSize;
-  private final int currentPage;
+  protected final int pageSize;
+  protected final Optional<Integer> currentPage;
+
+  public PaginationSpec(int pageSize) {
+    this.pageSize = pageSize;
+    this.currentPage = Optional.empty();
+  }
 
   public PaginationSpec(int pageSize, int currentPage) {
     this.pageSize = pageSize;
-    this.currentPage = currentPage;
+    this.currentPage = Optional.of(currentPage);
   }
 
   public int getPageSize() {
     return this.pageSize;
   }
 
   public int getCurrentPage() {
-    return this.currentPage;
+    return this.currentPage.get();
```

Comment:
This will throw an exception for the first ctor which is likely not desired or knowable by a caller.

Reply:
n/a having removed the extends relationship